zcash_client_backend/src/zip321.rs

```diff
@@ -106,7 +106,7 @@ impl Payment {
 /// payment value in the request.
 #[derive(Debug, PartialEq)]
 pub struct TransactionRequest {
-    payments: Vec<Payment>,
+    pub payments: Vec<Payment>,
```

Comment:
Actually, there is no restriction that applies once `Payment` values have been constructed except that a transaction request cannot contain more than 2109 payments; there is no restriction in the ZIP on creating multiple payments to the same address (intentionally) and the `Payment` type guarantees structural validity by construction. The 2109 payment limit might be worth enforcing this way, I guess.

Reply:
More than 2108 payments, since one Output is needed for change. Anyway, we should probably impose a lower limit than that; 2 Mbyte transactions are not particularly friendly.